### PR TITLE
Make `ModelExperimental.boundingSphere` update when invoked

### LINKED ISSUE #10626: Bounding sphere not updated right away after model matrix changes
This a small regression in `ModelExperimental`

If `modelMatrix` is changed it takes an update loop before `boundingSphere` is updated, unlike `Model` and `Cesium3DTileset` which update the `boundingSphere` on demand when the getter is called.

The difference between `Model` and `ModelExperimental` is shown in this sandcastle: [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bZJvb9MwEMa/yimvPClzhgZvtrQaFAQvmJgo4lWk6eJcGmuOHdlOu4L47vhPKzrWKIp8vruf78ljYbTzsJW0IwsL0LSDFTk5j/xn2mNNIVK8Mtqj1GSbooTfjQbYKNPSDfSoHDX6z8Vto0WiOUGaAixTeQpDstFVBasB9Ybg3nSkPj1PZOVIAazAm7wJqDuQHnbGPrkjcUyZRSbzKTRJL7fkOHYdO8z7Csl7a8bPyvcsjTtbdQNNwXkV3jWOk6KP6LFKbFdlyntpD6vHsOQb1Qa5UVxUlyq5Jez2DwEtHXE/kGb9rIWXRrNUcJH/Th7cW9ROYcyG8TOgNbPupN6sp4EscYudnN1t7Mn59L1Hb+Vz6DmqS/HbpOnHPyg7MWyF1ocV6mt2eXJuCVf8Kn2SiONkpu8d+ZeWfwnSwmAP0ovhe/SJXV4fekt4FwkJkF0QOJJFHk3+8EIROyezPBwYCemqNLooi9r5vaJlhMbnTo6TsT5axYJJnoJJGDRV7SyeyHPhXJ4AoK5OW+tObkF2izOXFYRC50Kmn5Vay1/UFMu6CvWvWpVJ6r9tySrcx7LhzfJr3uSc11UIz3d6Y1SL9j/yXw)

Peek 2022-08-02 18-04

### PR DESCRIPTION
**Pull request by j9liu**
Fixes #10626. This [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bZJvb9MwEMa/yimvPClzhgZvtrQaFAQvmJgo4lWk6eJcGmuOHdlOu4L47vhPKzrWKIp8vruf78ljYbTzsJW0IwsL0LSDFTk5j/xn2mNNIVK8Mtqj1GSbooTfjQbYKNPSDfSoHDX6z8Vto0WiOUGaAixTeQpDstFVBasB9Ybg3nSkPj1PZOVIAazAm7wJqDuQHnbGPrkjcUyZRSbzKTRJL7fkOHYdO8z7Csl7a8bPyvcsjTtbdQNNwXkV3jWOk6KP6LFKbFdlyntpD6vHsOQb1Qa5UVxUlyq5Jez2DwEtHXE/kGb9rIWXRrNUcJH/Th7cW9ROYcyG8TOgNbPupN6sp4EscYudnN1t7Mn59L1Hb+Vz6DmqS/HbpOnHPyg7MWyF1ocV6mt2eXJuCVf8Kn2SiONkpu8d+ZeWfwnSwmAP0ovhe/SJXV4fekt4FwkJkF0QOJJFHk3+8EIROyezPBwYCemqNLooi9r5vaJlhMbnTo6TsT5axYJJnoJJGDRV7SyeyHPhXJ4AoK5OW+tObkF2izOXFYRC50Kmn5Vay1/UFMu6CvWvWpVJ6r9tySrcx7LhzfJr3uSc11UIz3d6Y1SL9j/yXw) should now appear the same for `Model` and `ModelExperimental`.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/bounding-sphere-getter-fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Source/Scene/Model/Model.js [hunk 1]
```diff
@@ -2095,7 +2096,11 @@ function scaleInPixels(positionWC, radius, frameState) {
 function getScale(model, modelMatrix, frameState) {
   let scale = model.scale;
 
-  if (model.minimumPixelSize !== 0.0 && !model._projectTo2D) {
+  if (
+    defined(frameState) &&
```
Comment (lilleyse) [inline Source/Scene/Model/Model.js:2100]: This check feels a little awkward. Could `boundingSphere` and `computedScale` be updated separately? That way the `boundingSphere` setter would only need to update the bounding sphere which doesn't require `frameState`.

File: Source/Scene/Model/Model.js [hunk 2]
```diff
@@ -1102,6 +1102,7 @@ Object.defineProperties(Model.prototype, {
       }
       //>>includeEnd('debug');
 
+      updateBoundingSphereAndScale(this);
```
Comment (lilleyse) [inline Source/Scene/Model/Model.js:1105]: In `updateBoundingSphereAndScale` there's the line:

```
  if (!model._updateModelMatrix && model._minimumPixelSize === 0.0) {
    return;
  }
```

If `_updateModelMatrix` is false the bounding sphere won't be updated. Here's a sandcastle that shows when that could happen: [sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=zVNtb9MwEP4rVj6lUucMDb6MrIIVCT4wMdGKT5HQxbm0Fo5d2ZeOgvbfOdsZtFt/AFVl+d6e5+7xRTkbSOw1PqAXN8Lig1hi0OMgvyVf2RQq2UtnCbRF3xRz8buxQmyMa/Fa9GACNvZx9raxKqEFhRYZLKPKZHLwKTy4Dg2Hk1/uvB406T0GCV1XTuR3MUf23g0fDfVl4hu9uRZNIWXF/xUMO4MfgKBKeKHKle+1n27f+So3puV+Y3ez1EHKlR6hO9wzuA4oaYu27EerSDtbpoRZHjC36/o+IJ1q84kBtN3ca1Lbr2A3WF5cycu5uIzHGz4jm5gmVDCgBxnVuHWjjYWr3RY9ZjLZnjjnE2FEeMxNr8B2CgIZjBKtnTMt+NuRiPttirUHGwwQCk4THpmT+C0za3ysv7OdjEVTGQd4tnOdSM9DjiFNkuPpvAPy+ifXPD1Vsl+nx1r/Ay2P5FqCJ76BvSovjngnvaJcWa//Q/BiXtSBDgYXETL+3ulh5zzFDSx59wh591jwULWj+oEkVQiZX4i6Oi6tO70Xurs58xEJZSAEjvSjMSv9C5tiUVec/6LUuDT7lz16A4eYtn21+JydUsq6YvN8JeVVeYb8Bw)

**Comment by j9liu** [comment]
@lilleyse updated, could you take another look?

**Comment by ptrgags** [comment]
@j9liu updates look good, merging!

**Pull request merged**